src/ui/pages/PageManager.ts

```diff
@@ -0,0 +1,69 @@
+import { Application } from "pixi.js";
+import { BasePage } from "./BasePage";
+
+/**
+ * The controller class to manage the page navigation.
+ */
+export class PageManager {
+    private app: Application;
+    private currentPage: BasePage;
+    private currentPageIndex: number = -1;
+    private pages: Array<BasePage> = [];
```

Comment:
using an Object or map can be easier to manage in the future. like `{[pageId1]: Page1, [pageId2] : Page2}`

Reply:
makes sense, I'll review the approach we discussed...

Reply:
updated by https://github.com/irgat/soloTest/pull/3#discussion_r1721923203